packages/block-library/src/navigation-link/link-ui.js

```diff
@@ -56,85 +77,79 @@ export function getSuggestionsQuery( type, kind ) {
 	}
 }
 
-/**
- * Add transforms to Link Control
- *
- * @param {Object} props          Component props.
- * @param {string} props.clientId Block client ID.
- */
-function LinkControlTransforms( { clientId } ) {
-	const { getBlock, blockTransforms } = useSelect(
+function LinkUIBlockInserter( { clientId, onBack, onSelectBlock } ) {
+	const { rootBlockClientId } = useSelect(
 		( select ) => {
-			const {
-				getBlock: _getBlock,
-				getBlockRootClientId,
-				getBlockTransformItems,
-			} = select( blockEditorStore );
+			const { getBlockRootClientId } = select( blockEditorStore );
 
 			return {
-				getBlock: _getBlock,
-				blockTransforms: getBlockTransformItems(
-					_getBlock( clientId ),
-					getBlockRootClientId( clientId )
-				),
+				rootBlockClientId: getBlockRootClientId( clientId ),
 			};
 		},
 		[ clientId ]
 	);
 
-	const { replaceBlock } = useDispatch( blockEditorStore );
-
-	const featuredBlocks = [
-		'core/page-list',
-		'core/site-logo',
-		'core/social-links',
-		'core/search',
-	];
-
-	const transforms = blockTransforms.filter( ( item ) => {
-		return featuredBlocks.includes( item.name );
-	} );
+	const focusOnMountRef = useFocusOnMount( 'firstElement' );
 
-	if ( ! transforms?.length ) {
-		return null;
-	}
+	const dialogTitleId = useInstanceId(
+		LinkControl,
+		`link-ui-block-inserter__title`
+	);
+	const dialogDescritionId = useInstanceId(
+		LinkControl,
+		`link-ui-block-inserter__description`
+	);
 
 	if ( ! clientId ) {
 		return null;
 	}
 
 	return (
-		<div className="link-control-transform">
-			<h3 className="link-control-transform__subheading">
-				{ __( 'Transform' ) }
-			</h3>
-			<div className="link-control-transform__items">
-				{ transforms.map( ( item, index ) => {
-					return (
-						<Button
-							key={ `transform-${ index }` }
-							onClick={ () =>
-								replaceBlock(
-									clientId,
-									switchToBlockType(
-										getBlock( clientId ),
-										item.name
-									)
-								)
-							}
-							className="link-control-transform__item"
-						>
-							<BlockIcon icon={ item.icon } />
-							{ item.title }
-						</Button>
-					);
-				} ) }
-			</div>
+		<div
+			className="link-ui-block-inserter"
+			role="dialog"
+			aria-labelledby={ dialogTitleId }
+			aria-describedby={ dialogDescritionId }
+			ref={ focusOnMountRef }
+		>
+			<VisuallyHidden>
+				<h2 id={ dialogTitleId }>{ __( 'Add block' ) }</h2>
+
+				<p id={ dialogDescritionId }>
+					{ __( 'Choose a block to add to your Navigation.' ) }
+				</p>
+			</VisuallyHidden>
+
+			<Button
+				className="link-ui-block-inserter__back"
+				icon={ chevronLeftSmall }
+				onClick={ ( e ) => {
+					e.preventDefault();
+					onBack();
+				} }
+				size="small"
+				aria-label={ __( 'Return to link search dialog.' ) }
```

Comment:
I think we should remove the `aria-label` here so it relies on "Back". I don't think there's a good reason for this to diverge. By using an aria-label, it overwrites the "Back." 

Say I ask a blind screen reader user to test the "back button" on this component. They won't know which one I mean, because the button says "return to link search dialog." to them. It can create confusion for labels to diverge, so unless there's a clear reason, it's best for them to stay the same. I do like your description though :)

Reply:
@jeryj Do you have any bandwidth to implement these tweaks?